core/include/scipp/core/element/sort.h

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Thibault Chatel
+#pragma once
+
+#include "scipp/common/overloaded.h"
+#include "scipp/common/span.h"
+#include "scipp/core/element/arg_list.h"
+#include "scipp/core/transform_common.h"
+#include "scipp/core/value_and_variance.h"
+#include "scipp/units/unit.h"
+
+#include <stddef.h>
+
+namespace scipp::core::element {
+
+constexpr auto sort_common = overloaded{
+    core::element::arg_list<
+        std::tuple<span<int64_t>, span<int32_t>,
+        span<double>, span<float>,
+        span<std::string>>>,
+    [](units::Unit &) {}};
+
+constexpr auto sort_nondescending = overloaded{
```

Comment:
I think there must be a way to avoid having so much duplication here. The common way to change sort order in `std::sort` is to pass a different comparison operator (such as `std::less`, or a lambda) as a third argument, instead of using `rebin()` and `rend()`. Can you find a way to do this, maybe templating this sort helper functor? If not, I can give it a try.